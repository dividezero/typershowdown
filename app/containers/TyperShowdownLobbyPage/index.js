/*
 * TyperShowdownLobbyPage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { useState, useEffect } from 'react';
import { Pane } from 'evergreen-ui';
import GameListCard from '../../components/GameListCard';

import { getGameSessions } from '../../api/games';

const gameName = 'TyperShowdown';

export default function TyperShowdownLobbyPage() {
  const [gameList, setGameList] = useState([]);

  console.log('render');
  const refreshSessions = () => {
    getGameSessions(gameName).then(response => {
      response.text().then(body => {
        const { channels } = JSON.parse(body);
        console.log('channels', channels);
        setGameList(channels);
      });
    });
  };

  useEffect(() => {
    refreshSessions();
  }, []);

  return (
    <Pane padding={20}>
      <GameListCard
        gameList={gameList}
        onCreate={() => {}}
        onSelect={() => {}}
        onRefresh={refreshSessions}
      />
    </Pane>
  );
}
