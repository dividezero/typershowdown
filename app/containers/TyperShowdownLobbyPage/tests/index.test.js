import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import TyperShowdownLobbyPage from '../index';

describe('<TyperShowdownLobbyPage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <TyperShowdownLobbyPage />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
