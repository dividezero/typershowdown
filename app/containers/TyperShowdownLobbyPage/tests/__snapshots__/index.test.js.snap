// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TyperShowdownLobbyPage /> should render and match the snapshot 1`] = `
<h1>
  <span>
    This is the TyperShowdownLobbyPage container!
  </span>
</h1>
`;
